nitransforms/nonlinear.py

```diff
@@ -22,9 +31,24 @@ class DisplacementsFieldTransform(TransformBase):
     __slots__ = ["_field"]
 
     def __init__(self, field, reference=None):
-        """Create a dense deformation field transform."""
+        """
+        Create a dense deformation field transform.
+
+        Example
+        -------
+        >>> DisplacementsFieldTransform(test_dir / "someones_displacement_field.nii.gz")
+        <(57, 67, 56) field of 3D displacements>
+
+        """
         super().__init__()
-        self._field = np.asanyarray(field.dataobj)
+
+        field = _ensure_image(field)
+        self._field = np.squeeze(
+            np.asanyarray(field.dataobj) if hasattr(field, "dataobj") else field
+        )
+        self.reference = reference or field.__class__(
+            np.zeros(self._field.shape[:-1]), field.affine, field.header
```

Comment:
This will fail if `field` is an `ndarray`.

Reply:
Yes, perhaps this should be more explicit that field must be an image if reference is `None`. Will add an error